### PR TITLE
Fix: dry run from revision application patch error

### PR DESCRIPTION
**Pull request by LiuFang07**
Signed-off-by: old.prince <di7zhang@gmail.com>

### Description of your changes

Fixes:Dry-run from revision application,Problems caused by resource version lower than the current version

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/domain/service/application.go [hunk 1]
```diff
@@ -1513,6 +1513,13 @@ func (c *applicationServiceImpl) DryRunAppOrRevision(ctx context.Context, appMod
 		}
 	case "REVISION":
 		app, _, err = c.getAppModelFromRevision(ctx, appModel.Name, dryRunReq.Version)
+		originalApp := &v1beta1.Application{}
+		if err := c.KubeClient.Get(ctx, types.NamespacedName{
+			Name:      app.Name,
+			Namespace: app.Namespace,
+		}, originalApp); err == nil {
+			app.ResourceVersion = originalApp.ResourceVersion
```
Comment (barnettZQG) [inline pkg/apiserver/domain/service/application.go:1521]: If not set the resource version, what error will occur?

Comment (LiuFang07) [reply pkg/apiserver/domain/service/application.go:1521]: k8s will find that the current resource version is higher than the dryrun resource version, causing the update to fail.
detailed errors:cannot patch object: Operation cannot be fulfilled on applications.core.oam.dev \"demo\": the object has been modified; please apply your changes to the latest version and try again

Comment (StevenLeiZhang) [reply pkg/apiserver/domain/service/application.go:1521]: > k8s will find that the current resource version is higher than the dryrun resource version, causing the update to fail. detailed errors:cannot patch object: Operation cannot be fulfilled on applications.core.oam.dev "demo": the object has been modified; please apply your changes to the latest version and try again

Should we update the Dryrun version, when deploy Application? In order to keep Dryrun version follows the deployed Application

Comment (barnettZQG) [reply pkg/apiserver/domain/service/application.go:1521]: @LiuFang07  How to reproduce? Generating the application spec from the application config or revision should get a consistent result. So, if we need to set the resource version, both ways need to do it.

Comment (LiuFang07) [reply pkg/apiserver/domain/service/application.go:1521]: @barnettZQG 
reproduce:
curl --location --request POST 'http://example.doamin.com/api/v1/applications/policy/dry-run' \
--header 'Authorization: Bearer xxxxxx' \
--header 'Content-Type: application/json' \
--data-raw '{
    "dryRunType": "REVISION",
    "version": "20221230183820839" //not the currently running version
}'


Now,Generating the application spec from the application config,The renderOAMApplication method set the resource version.

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**